### PR TITLE
_posts: Add post about OpenSSL switch.

### PR DESCRIPTION
**Pull request by ericonr**

### CODE REVIEW

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:6]: This needs a date.  Also we're not planning on doing it, we will be performing the switch.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:10]: there should be no noticeable change, with the exception of a small handful of packages all your existing software will continue to work.  If your system has explicit dependencies on LibreSSL, you will of course need to take action to ensure your system continues to function after the switch.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:15]: Consider a horizontal rule here, and then a less flippant lead in to the technical discussion.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:19]: First use of term should be defined; link format should be `void-linux/void-packages#20935`.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:25]: This point reads weird, I think your qualifiers are in the wrong order.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:28]: This sounds like an aside, and should probably be formatted as such.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:29]: Dubious point here.  You should be prepared to back this up if its a claim you want to make.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:31]: You need a bridge paragraph here talking about Void not being the first to make this decision.  I'd talk about the fact that Alpine already dropped it, and we're reaching a point where Void is the last recognizable distro carrying libressl.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:37]: The next three paragraphs are kind of meandering and don't fit well.  You likely need a bridge between talking about other linux distros and talking about specific software on Void.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:55]: The rest of this article is written from the perspective of "The Void Team" and you switch to "I' here.  Keep the voice the article starts with.  The closing is from the entire Void team, as we have greatly benefited from their work.  Maybe make a mention here that we continue to use other fine software from OpenBSD such as openssh and signify?

**Review by the-maldridge — CHANGES_REQUESTED** [review]

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:24]: by the -> as a result of the / due to the

Perhaps a short statement about why it is risky (likelihood of breakage, etc).

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:23]: for patching packaged -> to patch (we won't be patching non-packaged software, lol)

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:21]: One of the main reasons which is not explicitly listed below is maintainer time/effort.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [reply _posts/2021-02-23-OpenSSL.markdown:25]: How about "Support for hardware accelerated hashing (? or is it encryption, i dont remember) on non-x86 architectures"

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:35]: This strengthens the argument when it is noted that LWN gave LibreSSL on Linux quite a poor review.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:40]: cyphers -> ciphers

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:38]: End the qt5 sentence, start a new one about need for rework in Qt6.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:39]: New paragraph?

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:47]: This reads that LibreSSL provides a bridging layer translating OpenSSL 1.0.1 style API to 1.1.1 which is not the case.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:49]: Perhaps unique instead of differentiating? Also note OpenSSL does not provide a `libtls` impl.

**Review by fosslinux — CHANGES_REQUESTED** [review]
I'm unsure if you want to write anything about how the reasons for the original switch to LibreSSL no longer apply, and that there are few benefits to keeping LibreSSL, as discussed in the RFC - but up to you.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (abenson) [reply _posts/2021-02-23-OpenSSL.markdown:19]: The discussion started in an
    [Request For Comments \(RFC\)](void-linux/void-packages/issues/20935)

File: _posts/2021-02-23-OpenSSL.markdown
Comment (abenson) [reply _posts/2021-02-23-OpenSSL.markdown:25]: "Better hardware acceleration on non-`x86-64` platforms"

File: _posts/2021-02-23-OpenSSL.markdown
Comment (abenson) [reply _posts/2021-02-23-OpenSSL.markdown:28]: Yeah, it doesn't really say anything -- why does OpenVPN get a callout that nothing else does?

File: _posts/2021-02-23-OpenSSL.markdown
Comment (abenson) [reply _posts/2021-02-23-OpenSSL.markdown:29]: I would replace this with something about less time spent on patching for LibreSSL compatible instead

File: _posts/2021-02-23-OpenSSL.markdown
Comment (abenson) [reply _posts/2021-02-23-OpenSSL.markdown:31]: I think Alpine dropping it was a bigger deal than Gentoo, since like Void, Alpine only offered the one option.  Gentoo always had the option of OpenSSL.  Also, Alpine dropping it had a bigger effect on Void, anyway.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (abenson) [reply _posts/2021-02-23-OpenSSL.markdown:37]: This could go as an example in my comment about time spent on patching.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (abenson) [reply _posts/2021-02-23-OpenSSL.markdown:6]: The Void Linux team is [switching back to OpenSSL](void-linux/void-packages/pull/21056)

File: _posts/2021-02-23-OpenSSL.markdown
Comment (abenson) [reply _posts/2021-02-23-OpenSSL.markdown:15]: I think dropping this line and replacing with a rule would be enough; the next sentence leads right on to the technical side of the discussion fairly naturally.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (abenson) [inline _posts/2021-02-23-OpenSSL.markdown:20]: This sounds choppy and run-on-y. maybe "and has continued with most maintainers expressing support."

File: _posts/2021-02-23-OpenSSL.markdown
Comment (abenson) [reply _posts/2021-02-23-OpenSSL.markdown:47]: Qt4 was already dropped, no need to bring it back up, even if this was a driving factor.

I'd replace it with a blurb about having to drop software that only works with the old API.  Are there are other packages that will need to be dropped? Or have they all been dropped already?

Comment (ahesford) [reply _posts/2021-02-23-OpenSSL.markdown:47]: Unfortunately, this move has required us to drop some packages that rely on the OpenSSL 1.0.1 API; while LibreSSL maintains compatibility with this legacy API, modern OpenSSL has abandoned it.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ahesford) [reply _posts/2021-02-23-OpenSSL.markdown:55]: The Void team is grateful for the work of the OpenBSD community on `[libressl-portable](https://github.com/libressl-portable/portable)`. Void continues to provide other excellent OpenBSD software, including OpenSSH and signify.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (Johnnynator) [reply _posts/2021-02-23-OpenSSL.markdown:28]: OpenVPN had breakage in the past due to LibreSSL, which was noticed by end users. So using it as example isn't bad imo. Furthermore it is used by a lot of people.

Comment (abenson) [reply _posts/2021-02-23-OpenSSL.markdown:28]: I think more detail about that would be a better example than just saying "it'll benefit" -- not everyone might remember the openvpn breakage.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (abenson) [reply _posts/2021-02-23-OpenSSL.markdown:55]: Maybe also include links to OpenSSH and especially signify

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:19]: @the-maldridge I'm not sure I understand what you mean by link format here. Something like `in an RFC, [void-linux/void-packages#20935](https://...)`?

Comment (the-maldridge) [reply _posts/2021-02-23-OpenSSL.markdown:19]: that's pretty much exactly what I mean.  the link should give you a good idea where it goes before you click on it.

Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:19]: @ahesford "ongoing" gives me the impression that the decision is still in progress, so I won't use it for now.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:28]: That's fair, I will add a link to it. I think it's important to mention, since it's a simple real world example, but it could be better explained.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:29]: That's one of the benefits as I understood it, but I am definitely not ready for that. Removing.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:20]: Thanks!

File: _posts/2021-02-23-OpenSSL.markdown
Comment (Gottox) [reply _posts/2021-02-23-OpenSSL.markdown:24]: Sounds nebolous, just say, that we had very poor performance when porting to ppc.

**Review by Gottox — COMMENTED** [review]
All in all okay, but I would remove the paragraph that talks about that other projects also abandoned libressl, if there's no relationship to our decision. Sounds like self-justification.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (mobinmob) [reply _posts/2021-02-23-OpenSSL.markdown:53]: >  Void continues to provide other excellent OpenBSD software, including OpenSSH and signify.

signify is not only provided in the repo, but used for signing the iso images.  It think it should be mentioned.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:24]: I went with `Extensive support for platform specific optimizations outside of x86.`

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:23]: The new paragraph I wrote may be too large, but I think it covers most of it.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:20]: I added another item for it, thanks.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:33]: I don't want to be too negative here

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:29]: I restructured it a lot, let me know what you think.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:23]: "to support LibreSSL" ?

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:25]: isses typo; perhaps results could be changed to something like problems?

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:30]: Rather than "that require it" (sounds ambiguous), how about "using SSL"? (is that correct?)

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:52]: This isn't very clear. The entire Qt6 thing can probably be shortened to "The new Qt6 would need significant effort to be patched and maintained to use LibreSSL". The first sentence is wordy, can probably be shortened to "For example, Void's version of the Qt toolkit ...."

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:76]: Keep in mind that this should probably be updated once live to say that work has been done, once that PR gets merged.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (fosslinux) [inline _posts/2021-02-23-OpenSSL.markdown:80]: Perhaps provide should be package?

**Review by fosslinux — CHANGES_REQUESTED** [review]
I like this a lot more than the previous version, only a few things I picked up.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (Gottox) [inline _posts/2021-02-23-OpenSSL.markdown:83]: I think the article shouldn't end with offtopic. It should summarize everything instead. Maybe add something like "We hope that our switch to OpenSSL will ease maintainance and give you access to newer crypto standards and cyphers."

**Review by Gottox — APPROVED** [review]
For me it's fine. Just one suggestion.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:25]: "resulting problems" :p 

And I added a bit more to say that we don't know of any security issues caused by our patches.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:30]: It does, so I switched to "OpenSSL-like library", not sure it's best.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:52]: This makes me lose the tie in into the joke about Qt receiving updates, but I can live with it :P

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown:76]: I will say work has been done, since @Johnnynator pulled that into the main OpenSSL PR.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ahesford) [inline _posts/2021-02-23-OpenSSL.markdown:20]: A discussion about switching to OpenSSL began in a Request For Comments (RFC) posted on April 12th, 2020, at `[void-linux/void-packages#20935](https://github.com/void-linux/void-packages/issues/20935)`. Since then, a majority of Void maintainers have expressed support for the move.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ahesford) [inline _posts/2021-02-23-OpenSSL.markdown:24]: The complexity of the OpenSSL API makes patching burdensome and risky, with mistakes potentially causing runtime errors or security issues.

**Review by ahesford — APPROVED** [review]
I offer a few more refinements to consider, but this should probably be buttoned up shortly to provide as much advance notice as possible.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:63]: which is limited in the ciphers available to running

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:67]: , which we have received but reports for

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:70]: s/which resulted/resulting/

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:81]: a hard to misuse interface.

As written this is a fragment.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:88]: Not sure this last sentence on signify belongs here.  It links elsewhere and dilutes the focus of the post.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [inline _posts/2021-02-23-OpenSSL.markdown:94]: Very abrupt, needs another sentence after it to the effect of "look forward to the update landing some time after March 5th"

**Review by the-maldridge — CHANGES_REQUESTED** [review]

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown]: I just copied your sentence :)

File: _posts/2021-02-23-OpenSSL.markdown
Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown]: Mentioned the signing part, but without the link or anything.

File: _posts/2021-02-23-OpenSSL.markdown
Comment (the-maldridge) [reply _posts/2021-02-23-OpenSSL.markdown]: you did, I thought this was going at the end of a larger paragraph, or at the start of one.

Comment (ericonr) [reply _posts/2021-02-23-OpenSSL.markdown]: I meant that I copied this new sentence onto that paragraph :p 

which was indeed already a sentence you had suggested previously

**Review by the-maldridge — APPROVED** [review]

**Review by Johnnynator — APPROVED** [review]

**Pull request merged**
